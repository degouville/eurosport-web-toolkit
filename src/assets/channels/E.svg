<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 23">
  <path fill="#B9BBCA" fill-rule="evenodd" d="M10.335 6.656H5.346V.96c0-.406.33-.737.733-.737h13.01c.404 0 .732.333.732.74v3.192c0 .44-.273.755-.707.755h-8.78v1.746zm-5.166 16.25l5.166-4.685h8.764c.42 0 .724.36.724.73v3.219c0 .406-.292.736-.696.736H5.17zm5.166-8.909l6.62-2.836h-6.62V6.656L6.499 11.16H0l4.15 2.836L.586 18.22l5.951-2.594 3.797 2.594z"/>
</svg>
