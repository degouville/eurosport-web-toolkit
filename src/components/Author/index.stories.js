import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { Author } from '../..';

const authorStories = storiesOf('Author', module);

authorStories
  .add(
    'Author with picture',
    withInfo()(() => (
      <Author
        name={text('Name', 'Author name')}
        img={text('Avatar', 'https://i.eurosport.com/2014/04/23/1223468-25519942-150-150.jpg')}
        time={text('Time', 'September 4th 2018 at 12:40PM')}
      />
    ))
  )
  .add(
    'Author without picture',
    withInfo()(() => <Author name={text('Name', 'Author name')} time={text('Time', 'September 4th 2018 at 12:40PM')} />)
  );
