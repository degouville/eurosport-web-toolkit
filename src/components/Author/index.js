import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import E from '../../assets/channels/E.svg';
import { colors, theme, breakpoints } from '../..';

const StyledAuthor = styled.div`
  display: flex;
  align-items: center;
`;

const StyledImage = styled.img`
  height: 32px;
  width: 32px;
  ${breakpoints.small(css`
    height: 40px;
    width: 40px;
  `)};
`;

const StyledSpacer = styled.div`
  width: 1px;
  height: 48px;
  background: ${colors.white};
  opacity: 0.3;
  margin: 0 18px 0 11px;
  ${breakpoints.small(css`
    height: 64px;
    margin: 0 17px 0 13px;
  `)};
  ${breakpoints.small(css`
    height: 56px;
    margin: 0 33px 0 26px;
  `)};
`;

const StyledContent = styled.div`
  flex: 1;
  font-family: ${theme.typography.base};
  font-size: 14px;
  line-height: 24px;
  ${breakpoints.small(css`
    font-size: 16px;
  `)};
`;

const StyledName = styled.p`
  color: ${colors.white};
  font-weight: bold;
`;

const StyledTime = styled.p`
  color: ${colors.manatee};
`;

const Author = ({ name, img, time, ...props }) => (
  <StyledAuthor {...props}>
    <StyledImage src={img} alt={name} />
    <StyledSpacer />

    <StyledContent>
      <StyledName>{name}</StyledName>
      <StyledTime>{time}</StyledTime>
    </StyledContent>
  </StyledAuthor>
);

Author.propTypes = {
  name: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  img: PropTypes.string,
};

Author.defaultProps = {
  img: E,
};

export default Author;
