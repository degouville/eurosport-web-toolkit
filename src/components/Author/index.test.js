import { shallow, mount } from 'enzyme/build';
import React from 'react';
import E from '../../assets/channels/E.svg';

import Author from '.';

describe('Author', () => {
  it('renders Author', () => {
    expect(shallow(<Author name="name" img="img" time="time" />)).toMatchSnapshot();
  });
  it('should spread props', () => {
    const wrapper = shallow(<Author name="name" time="time" data-test-id="test" />);
    expect(wrapper.prop('data-test-id')).toEqual('test');
  });
  it('renders Author without avatar', () => {
    const wrapper = mount(<Author name="name" time="time" />);
    expect(wrapper.prop('img')).toEqual(E);
  });
});
