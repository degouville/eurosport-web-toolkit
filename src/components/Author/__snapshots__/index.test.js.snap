// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders Author 1`] = `
<Styled(div)>
  <Styled(img)
    alt="name"
    src="img"
  />
  <Styled(div) />
  <Styled(div)>
    <Styled(p)>
      name
    </Styled(p)>
    <Styled(p)>
      time
    </Styled(p)>
  </Styled(div)>
</Styled(div)>
`;
